File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let result = 1;
    for(i = 2; i <= number; i++)
    {
        let count = 2;
        let temp = Math.sqrt(i);
        for(let j = 2; j <= temp; j++)
        {
            if(i % j === 0)
            {
                if(temp === j) count++;
                else count += 2;
            }
        }
        if(limit < count) result += power
        else result+= count
    }
    
    return result
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cs
using System;
using static System.Math;

public class Solution {
    public int solution(int number, int limit, int power) {
        int need = 0;
        
        for(int i = 1; i <= number;i++)
        {
            int count = 0;
            for(int j = 1; j <= Math.Sqrt(i); j++)
            {
                if(i % j == 0)
                {
                    count += i / j == j ? 1 : 2;
                }
            }
            need += count > limit ? power : count;
        }
        return need;
    }
}